### PR TITLE
teamcity: add volumePermissions for non root containers usage

### PR DESCRIPTION
**Pull request by 3rmack**

### CODE REVIEW

File: charts/teamcity/templates/agent-daemonset.yaml [hunk 1]
```diff
@@ -38,6 +38,39 @@ spec:
...
+            cpu: 50m
+            memory: 32Mi
+          requests:
+            cpu: 50m
+            memory: 32Mi
+        command:
+          - sh
+          - -c
+          - |
+            chown -Rv {{ .Values.securityContext.runAsUser }}:{{ .Values.securityContext.runAsGroup }} /data
```
Comment (Strangerxxx) [inline charts/teamcity/templates/agent-daemonset.yaml]: no multiline needed

File: charts/teamcity/values.yaml [hunk 2]
```diff
@@ -86,6 +86,15 @@ securityContext: {}
...
 
+volumePermissions: {}
+#  enabled: true
+#  image:
+#    image: busybox
+#    imageTag: latest
+#    pullPolicy: IfNotPresent
+#  securityContext:
+#    runAsUser: 0
+
```
Comment (Strangerxxx) [inline charts/teamcity/values.yaml]: no need to manage it so precise
kind of simple `fixPermissions: true/false` should be enough

File: charts/teamcity/templates/server-deployment.yaml [hunk 3]
```diff
@@ -29,6 +29,31 @@ spec:
...
+              cpu: 50m
+              memory: 32Mi
+            requests:
+              cpu: 50m
+              memory: 32Mi
+          command:
+            - sh
+            - -c
+            - |
+              chown -Rv {{ .Values.agent.securityContext.runAsUser }}:{{ .Values.agent.securityContext.runAsGroup }} /data
```
Comment (Strangerxxx) [inline charts/teamcity/templates/server-deployment.yaml]: no need to have it multiline

File: charts/teamcity/templates/agent-daemonset.yaml [hunk 4]
```diff
@@ -38,6 +38,39 @@ spec:
...
+          - name: buildagent-temp
+            mountPath: /data/temp
+          - name: buildagent-tools
+            mountPath: /data/tools
+          - name: buildagent-plugins
+            mountPath: /data/plugins
+          - name: buildagent-system
+            mountPath: /data/system
+        securityContext:
+          runAsUser: {{ .Values.volumePermissions.securityContext.runAsUser }}
```
Comment (Strangerxxx) [inline charts/teamcity/templates/agent-daemonset.yaml]: it should run as root in any case to fix the permissions

File: charts/teamcity/templates/agent-daemonset.yaml (hunk 1 above)
Comment (Strangerxxx) [inline charts/teamcity/templates/agent-daemonset.yaml]: shouldn't we follow symlinks with `-h`?

File: charts/teamcity/templates/server-deployment.yaml [hunk 5]
```diff
@@ -29,6 +29,31 @@ spec:
...
+          volumeMounts:
+            - name: data
+              mountPath: /data
+            - name: artifacts
+              mountPath: /data/system/artifacts
+          securityContext:
+            runAsUser: {{ .Values.volumePermissions.securityContext.runAsUser }}
```
Comment (Strangerxxx) [inline charts/teamcity/templates/server-deployment.yaml]: it should run as root in any case to fix the permissions

**Review by Strangerxxx — CHANGES_REQUESTED** [review]

File: charts/teamcity/templates/agent-daemonset.yaml (hunk 1 above)
Comment (3rmack) [reply charts/teamcity/templates/agent-daemonset.yaml]: as far as I saw there are no symlinks in these dirs

**Review by Strangerxxx — APPROVED** [review]
lgtm

**Pull request merged**